src/relay/backend/graph_executor_codegen.cc

```diff
@@ -215,24 +215,35 @@ class GraphExecutorCodegen : public backend::MemoizedExprTranslator<std::vector<
 
     if (memory_plan_.defined()) {
       // TODO(@electriclilies, @jroesch): remove UpdateMainWorkspaceSize
-      func_info =
-          relay::tec::UpdateMainWorkspaceSize(mod, targets_, memory_plan_->expr_to_storage_info);
+      tec::TargetMap tec_target_map;
+      for (const auto& pair : targets_) {
+        tec_target_map.emplace(static_cast<DLDeviceType>(pair.first->value), pair.second);
+      }
+      func_info = relay::tec::UpdateMainWorkspaceSize(mod, tec_target_map,
```

Comment:
Do we have a tracking issue for flowing this into the actual lowering machinery?

Reply:
CORE-122